stressTestLog/send_fake_data_py3.py

```diff
@@ -0,0 +1,363 @@
+        return f"--- Check project {project_name} failed ---"
+
+    # all data
+    data = []
+    count = 0
+    send_data_time = time.time()
+    threads = []
+    for i in range(1, config_vars['instance_count'] + 1):
+        utc_time_now = int(arrow.utcnow().float_timestamp) * 1000
+        instance = f"{config_vars['instance_prefix']}{i}"
+        for x in range(0, log_count_per_instance):
+            entry = {
+                "tag": instance,
+                "eventId": utc_time_now,
+                "data": choice(SAMPLE_LOGS)
+            }
+            data.append(entry)
+            count += 1
+
+            # use number 1000 for check the size
+            if count % 1000 == 0 and get_json_size_bytes(data) >= config_vars['chunk_size']:
+                # send data
+                # send_data(config_vars, project_name, data)
+                loop_thead = Thread(target=send_data, args=(config_vars, project_name, data))
+                loop_thead.start()
+                threads.append(loop_thead)
+                data = []
+
+    if len(data) != 0:
+        # send_data(config_vars, project_name, data)
+        loop_thead = Thread(target=send_data, args=(config_vars, project_name, data))
+        loop_thead.start()
+        threads.append(loop_thead)
+
+    for thread in threads:
+        thread.join()
+
+    return f"--- Total count for {project_name}: {count}. Use {round(time.time() - send_data_time, 2)} seconds ---"
+
+
+def handle_send_data(config_vars):
+    logs_total_count = config_vars['logs_total_count']
+    project_list = config_vars['project_list']
+    log_count_per_instance = int(logs_total_count / len(project_list) / config_vars['instance_count'])
+
+    # get args
+    arg_list = [(config_vars, project_name, log_count_per_instance) for project_name in project_list]
+
+    # start sub process by pool
+    print(f"--- Start of send log data at {arrow.utcnow()} ---")
+    try:
+        pool = Pool(min(len(project_list), multiprocessing.cpu_count()))
+        pool_result = pool.map_async(handle_send_data_project, arg_list)
+        pool.close()
+
+        results = pool_result.get()
+        pool.join()
+
+        for result in results:
+            print(result)
+    except KeyboardInterrupt:
+        print('(%s) main exception' % os.getpid())
+
+    print(f"--- End of send log data at {arrow.utcnow()} ---")
+
+
+def int_handler(signum, frame):
+    print('(%s) int_handler' % os.getpid())
+    raise KeyboardInterrupt()
+
+
+def main():
+    # monitor SIGTERM
+    signal.signal(signal.SIGTERM, int_handler)
+
+    urllib3.disable_warnings(urllib3.exceptions.InsecureRequestWarning)
+    importlib.reload(sys)
+    config_vars = get_agent_config_vars()
+
+    while True:
+        buffer_check_thead = Thread(target=handle_send_data,
+                                    args=(config_vars,))
+        buffer_check_thead.start()
+        buffer_check_thead.join()
+        # run with 1 minute interval
+        time.sleep(60)
```

Comment:
This should be a variable sleep based on how much time is left in the minute after sending the data. Otherwise, this will send the data, wait for a minute and then send the data again.

Reply:
Possible: 
next = now() + 1m 

buffer_check_thread -> run 

if (next > now()) 
time.sleep(next - now()) 
else 
print run took over a minute to send data

Reply:
> This should be a variable sleep based on how much time is left in the minute after sending the data. Otherwise, this will send the data, wait for a minute and then send the data again.

Will not wait thread join.